### DISCUSSION TITLE
focus an input element when `Show` becomes active

### DISCUSSION THREAD

**Discussion by M1cha** [leptos-rs/leptos#658] [General]
When first rendering the element, the browser can set focus itself if I add `autofocus` to the element, but since I open the `input` conditionally using the `Show` component I need a way to do that through leptos as well.

As far as I can tell there's no "element got attached to the DOM" event in leptos.
I've also tried to switch from `Show` to a closure returning a `view!` conditionally expecting that'd make leptos re-render the element every time the condition changes. And while the render function is getting called, the `autofocus` attribute doesn't seem to cause a re-focus the second time.

Maybe leptos never actually creates a new `input` and just applies the difference instead?
If so, do you have a suggestion for how to implement this seemingly simple functionality?

**Comment by gbj** [comment]
Could you share some example code to look at?

**Reply by M1cha** [reply]
of course. I create a full example project(CSR, so just use `trunk serve`): https://github.com/M1cha/leptos-experiments/tree/input-focus

[This](https://github.com/M1cha/leptos-experiments/blob/f9184e13c2b6cee9b56c1a9661c93f403ddeba33/src/lib.rs#L20) is the place where I have to work around the lack of a DOM event using a timer.

**Comment by gbj** [comment]
`HtmlElement<_>` has an `on_mount` function that does exactly what you want. Here's the idiomatic way to do what you're trying to do, I think:
```rust
create_effect(cx, move |_| {
    if show.get() {
        if let Some(ref_input) = ref_input.get() {
            ref_input.on_mount(|input| {
                input.focus();
            });
        }
    }
});
```

**Reply by M1cha** [reply]
oh how did I miss that function :blush: 
That works great, thanks a lot :tada:

**Reply by ja-he** [reply]
Since `on_mount` seems to be gone since v0.7.0 (with #2607 I believe) I'm wondering what the solution would be now?

(Hope it's OK to ask this here where some searching led me; happy to ask in a new discussion as well. My use case is how the charming crate's WasmRenderer wants to render to a div with an ID, meaning the div has to be mounted.)
